### ISSUE TITLE
indentation/alignment in jsx (react/jsx-curly-newline)

### ISSUE THREAD

**Issue by havenchyk** [standard/standard#1458]
**What version of this package are you using?**

eslint-config-standard-jsx@8.1.0

**What operating system, Node.js, and npm version?**
MacOS Catalina, node 13.2.0, npm 6.13.1
**What happened?**
Weird formatting, in prettier-standard package it's done more "correctly"
**What did you expect to happen?**
Another formatting like it's done in prettier-standard
**Are you willing to submit a pull request to fix this bug?**
Nope :(

I came across some conflict between `prettier-standard` and eslint with `standard` and `standard-jsx` presets included. 

Example:

```
const A: React.FC = ({ children }) => <div>{children}</div>
const B: React.FC<{ onDelete: () => void }> = ({ children }) => (
  <div>{children}</div>
)

const MyComponent: FunctionComponent = () => {
  const someLongMethodHandlerName = (obj: {
    parameter1: string
    parameter2: string
  }) => {}

  return (
    <A>
      <B
        onDelete={() =>
          someLongMethodHandlerName({
            parameter1: 'parameter1',
            parameter2: 'parameter2'
          })
        }
      >
        B
      </B>
    </A>
  )
}
```

^^ this is how prettier-standard formats the code,

but eslint with `standard` and `standard-jsx` presets expects the next version

```diff
const A: React.FC = ({ children }) => <div>{children}</div>
const B: React.FC<{ onDelete: () => void }> = ({ children }) => (
  <div>{children}</div>
)

const MyComponent: FunctionComponent = () => {
  const someLongMethodHandlerName = (obj: {
    parameter1: string
    parameter2: string
  }) => {}

  return (
    <A>
      <B
        onDelete={() =>
          someLongMethodHandlerName({
            parameter1: 'parameter1',
            parameter2: 'parameter2'
-         })
-       }
+ .       })}
      >
        B
      </B>
    </A>
  )
}
```

To me first variant is more correct /cc @sheerun

**Comment by sheerun** [comment]
This probably all stems from fast that `standard` uses eslint for both whitespace correction and linting issues in code. Eslint is great at linting, but not so much at correcting whitespace. prettier-standard uses [prettierx](https://github.com/brodybits/prettierx) for formatting and checking whitespace which can be configured to `standard` rules. I secretly hope standard will adopt this technique as well, so I can deprecate prettier-standard and point people here.

**Comment by havenchyk** [comment]
I would provide a PR if this proposal is approved /cc @feross

**Comment by feross** [comment]
I agree that this is a bug. This is how it should be formatted (removed the typescript types):

```js
const A = ({ children }) => <div>{children}</div>
const B = ({ children }) => (
  <div>{children}</div>
)

const MyComponent = () => {
  const someLongMethodHandlerName = (obj) => {}

  return (
    <A>
      <B
        onDelete={() =>
          someLongMethodHandlerName({
            parameter1: 'parameter1',
            parameter2: 'parameter2'
          })
        }
      >
        B
      </B>
    </A>
  )
}
```

**Comment by feross** [comment]
Hmm, this seems to be caused by the way the `react/jsx-curly-newline` rule works. Essentially, because the arrow function starts immediately after the open brace of of the JSX attribute (e.g. `onDelete={() =>`) the rule sees it as a single-line attribute and expects the closing brace to be on the same line when the arrow function ends.

This may be a bug in the rule. Or perhaps a new option could be added to support this?

If you'd like to see this fixed, I'd inquire about getting this fixed upstream in https://github.com/yannickcr/eslint-plugin-react

**Comment by LeonxLJX** [comment]
I'd like to take this one (`indentation/alignment in jsx (react/jsx-curly-newline)`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
I'd like to take this one (`indentation/alignment in jsx (react/jsx-curly-newline)`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)
